### PR TITLE
#4: Countdown Section

### LINKED ISSUE #4: Create the countdown text
Screen Shot 2021-02-04 at 09 36 59

There should be a countdown text for the event

- [x] The text should dynamically be updated
- [x] Text should follow the design
- [x] Check how it looks like in smaller screens

### PR DESCRIPTION
**Pull request by bgoktugozdemir**
* Section was created.
* AppBar was removed.
* The countdown text set dynamically.
* All the texts were created with ResponsiveBuilder.

Fixes #4 

Full Size | Mobile
------------ | -------------
Screen Shot 2021-02-08 at 16 55 46 | Screen Shot 2021-02-08 at 16 56 31

### CODE REVIEW

File: lib/widgets/countdown_section.dart [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+  void dispose() {
+    _timer.cancel();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final currentDate = DateTime.now();
+    final remainingTime = _eventStartingDate.difference(currentDate);
+    final text = '<bold>${remainingTime.inDays}</bold> gün '
```
Comment (salihgueler) [inline lib/widgets/countdown_section.dart:91]: Why don't you use RichText here?

File: lib/widgets/countdown_section.dart [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        '<bold>${remainingTime.inHours % 24}</bold> saat '
+        '<bold>${remainingTime.inMinutes % 60}</bold> dakika '
+        '<bold>${remainingTime.inSeconds % 60}</bold> saniye kaldı';
+
+    return _CountdownText(text: text);
+  }
+
+  void _setTimer() {
+    _timer = Timer.periodic(
+      const Duration(seconds: 1),
```
Comment (salihgueler) [inline lib/widgets/countdown_section.dart:101]: Why don't we use streams for this with a stream builder instead of a setState?

**Review by salihgueler — CHANGES_REQUESTED** [review]

File: lib/widgets/countdown_section.dart (hunk 2 above)
Comment (bgoktugozdemir) [reply lib/widgets/countdown_section.dart:101]: Do we need this? It reloads only the text.

File: lib/widgets/countdown_section.dart (hunk 1 above)
Comment (bgoktugozdemir) [reply lib/widgets/countdown_section.dart:91]: Because, the tag highlighter package is doing all these works for me.

Comment (salihgueler) [reply lib/widgets/countdown_section.dart:91]: But that is wrong :) you just pass in markdown and rely on library for something that you can do out of box.

File: lib/widgets/countdown_section.dart (hunk 2 above)
Comment (salihgueler) [reply lib/widgets/countdown_section.dart:101]: With StreamBuilder your control over the state update is more performant. On the other hand doing setState with each tick might be a problem. Plus, StreamBuilder would be less code and StatefulWidget can turn into Stateless ;)

File: lib/widgets/countdown_section.dart (hunk 1 above)
Comment (bgoktugozdemir) [reply lib/widgets/countdown_section.dart]: Done.

File: lib/widgets/countdown_section.dart (hunk 2 above)
Comment (bgoktugozdemir) [reply lib/widgets/countdown_section.dart]: Fixed.

**Comment by bgoktugozdemir** [comment]
It cherry-picked on #44

**Pull request closed without merging**